internal/log/logger.go

```diff
@@ -0,0 +1,30 @@
+// Package log defines the logger interface used in rsync library.
+package log
+
+import "log"
+
+// Logger logs messages.
+type Logger interface {
+	// Printf logs message to the underlaying log output. Arguments are handled in the manner of fmt.Printf.
```

Comment:
nit: underlying (no “a”), as we need an adjective here (see also https://wikidiff.com/underlying/underlaying)